File: Snooper/src/androidTest/java/com/prateekj/snooper/formatter/PlainTextFormatterTest.java
package com.prateekj.snooper.formatter;

import org.junit.Test;

import static com.prateekj.snooper.utils.TestUtilities.readFrom;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class PlainTextFormatterTest {

  @Test
  public void shouldReturnFormattedPlainText() throws Exception {
    ResponseFormatter formatter = new PlainTextFormatter();
    String rawString = readFrom("person_details_formatted_response.txt").replaceAll("\n", "\r");
    String formattedResponse = formatter.format(rawString);
    String expectedResponse = readFrom("person_details_formatted_response.txt");

    assertThat(formattedResponse, is(expectedResponse));
  }
}

File: Snooper/src/androidTest/java/com/prateekj/snooper/formatter/PlainTextFormatterTest.kt
package com.prateekj.snooper.formatter

import org.junit.Test

import com.prateekj.snooper.utils.TestUtilities.readFrom
import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat

class PlainTextFormatterTest {

  @Test
  @Throws(Exception::class)
  fun shouldReturnFormattedPlainText() {
    val formatter = PlainTextFormatter()
    val rawString =
      readFrom("person_details_formatted_response.txt").replace("\n".toRegex(), "\r")
    val formattedResponse = formatter.format(rawString)
    val expectedResponse = readFrom("person_details_formatted_response.txt")

    assertThat(formattedResponse, `is`<String>(expectedResponse))
  }
}